File: codegen/codegen-erlang/src/test/resources/ir/handler_discovery_dispatch_handler.expected.erl
dispatch_handler(Fun, Ctx, Input, Meta) ->
    Handlers = persistent_term:get(?HANDLERS_KEY, #{}),
    case maps:get(Fun, Handlers, undefined) of
        Handler when is_function(Handler, 3) ->
            Handler(Ctx, Input, Meta);
        _ ->
            {error, not_implemented}
    end.

File: codegen/codegen-elixir/src/test/resources/ir/handler_discovery_dispatch_handler.expected.ex
defp dispatch_handler(fun, ctx, input, meta) do
  handlers = :persistent_term.get(@handlers_key, %{})
  case Map.get(handlers, fun) do
    handler when is_function(handler, 3) -> handler.(ctx, input, meta)
    _ -> {:error, :not_implemented}
  end
end
